pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -226,3 +230,24 @@ func buildGatewayInboundHTTPRouteConfig(env model.Environment, gatewayName strin
 		VirtualHosts: virtualHosts,
 	}
 }
+
+func buildGatewayInboundNetworkFilters(env model.Environment, server *networking.Server) []listener.Filter {
+	filters := make([]listener.Filter, 0, len(server.Hosts))
+	for _, host := range server.Hosts {
```

Comment:
@rshriram This feels wrong I think. The server is a TCP port, so all hosts must be TCP services; enumerate the services in the registry by host name, and wire up network filters for each instance(? this feels like it should be per-service, not per instance-of-service).

I'm wiring up some tests shortly...

Reply:
this is missing the binding to virtual service. you might want to reuse the code that selects all virtual services bound to this gateway, and then run through the tcp part of virtual service spec, match server.port to tcp.match.port if present, and then generate the upstream cluster name from the destination. Only then do we do `buildOutboundNetworkFilters(clusterName, address, servicePort[which has protocol])`. 
But this raises the question of what on earth is the gateway.server.port.Protocol is useful for, in TCP context?

Reply:
Put another way, this is doing something that CF folks wanted which is to impliclity bind gateway port to all internal services matching that port/protocol